### PR TITLE
#11544: Replace tt_lib with ttnn in mobilenet, nanogpt, mnist

### LINKED ISSUE #11544: Replace tt_lib in mobilenet, nanogpt, mnist
Issue : #11423 , Tracking Issue : #11240 , Master Issue : #10532 

Update the following folders : 

- [x] mnist
- [x] mobilenet
- [x] nanogpt

Current status: Ready to merge

### PR DESCRIPTION
**Pull request by Aswinmcw**
Tracking Issue : #11544 

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: models/experimental/nanogpt/tt/nanogpt_attention.py [hunk 1]
```diff
@@ -118,7 +108,7 @@ def forward(self, x: tt_lib.tensor.Tensor) -> tt_lib.tensor.Tensor:
         tt_y = ttnn.matmul(tt_att, v)
 
         tt_y = ttnn.transpose(tt_y, 1, -2)
-        tt_y = tt_lib.tensor.reshape(tt_y, 1, B, T, C)
+        tt_y = ttnn.reshape(tt_y, 1, B, T, C)
```
Comment (KalaivaniMCW) [inline models/experimental/nanogpt/tt/nanogpt_attention.py:111]: use ttnn.experimental. 
see sheet

Comment (KalaivaniMCW) [reply models/experimental/nanogpt/tt/nanogpt_attention.py:111]: for all reshape occurrences

**Review by ayerofieiev-tt — APPROVED** [review]

**Comment by ayerofieiev-tt** [comment]
keep in mind, there is PR merging reshape into ttnn

File: models/experimental/nanogpt/nanogpt_utils.py [hunk 2]
```diff
@@ -41,21 +41,21 @@ def cache_weights_in_weka(device, dtype, reset_seeds):
...
+            value = ttnn.Tensor(
                 value.reshape(-1).tolist(),
                 value.shape,
                 weights_dtype,
-                tt_lib.tensor.Layout.ROW_MAJOR,
-            ).to(tt_lib.tensor.Layout.TILE)
-        tt_lib.tensor.dump_tensor(file_name + str(key) + str(weights_dtype) + ".bin", value)
+                ttnn.ROW_MAJOR_LAYOUT,
+            ).to(ttnn.TILE_LAYOUT)
+        ttnn.experimental.tensor.dump_tensor(file_name + str(key) + str(weights_dtype) + ".bin", value)
```
Comment (KalaivaniMCW) [inline models/experimental/nanogpt/nanogpt_utils.py:58]: can use ttnn.dump_tensor instead

**Review by KalaivaniMCW — APPROVED** [review]
Approved. please include ttnn.dump_tensor change before merging.

**Review by eyonland — APPROVED** [review]

**Pull request merged**